frontend/src/components/modules/GameInterface/GameInterface.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useState, useEffect } from "react";
+import { Typography, Box } from "@mui/material";
+import PointTable from "./PointTable";
+import Timer from "./Timer";
+import OfficialButtons from "./OfficialButtons";
+import TimerButton from "./TimerButton";
+import api from "api/axios";
+import { useParams } from "react-router-dom";
+import { type AddPointRequest } from "types/requests";
+import type { PointType, PlayerColor, Match, MatchPlayer } from "types/models";
+import "./GameInterface.css";
+import { useAuth } from "context/AuthContext";
+import { joinMatch, leaveMatch } from "sockets/emit";
+import { useSocket } from "context/SocketContext";
+
+export interface matchData {
+  timerTime: number;
+  players: MatchPlayer[];
+  playerNames: string[];
+  winner: string | undefined;
+  officials: string;
+}
+
+const GameInterface: React.FC = () => {
+  const [matchInfo, setMatchInfo] = useState<matchData>({
+    timerTime: 300,
+    players: [],
+    playerNames: [],
+    winner: undefined,
+    officials: ""
+  });
+
+  useEffect(() => {
+    if (matchId !== undefined) {
+      joinMatch(matchId);
+
+      return () => {
+        leaveMatch(matchId);
+      };
+    }
+  }, []);
+
+  const [open, setOpen] = useState(false);
+  const [selectedButton, setSelectedButton] = useState<string>("");
+  const [timer, setTimer] = useState<number>(matchInfo.timerTime);
+  const [isTimerRunning, setIsTimerRunning] = useState<boolean>(false);
+  const [playerColor, setPlayerColor] = useState<PlayerColor>("red");
+
+  const { matchId } = useParams();
+  const { userId } = useAuth();
+  const { matchInfo: matchInfoFromSocket } = useSocket();
+
+  useEffect(() => {
+    const getMatchData = async (): Promise<void> => {
```

Comment:
We shouldn't have to fetch the userInfo for the players. Since the matches are under the tournament routes (or rather should be), you can infer the player info from the current tournament route, i.e., accessing the tournament via `useTournament`. 

In order for this to work you have to move the route under the singular tournaments route. You can then access the provided hook.

One thing to note however, is that even if you move this under the tournament round, you probably wont have access to the hook since the context that the tournamentProvider passes doesnt go through the authProvider. 

I will open up a PR about this so that we can implement this more neatly